lib/traject/config/geo_config.rb

```diff
@@ -127,7 +127,15 @@ def mods_display(method, *args, default: nil)
     accumulator << 'Restricted'
   end
 end
-to_field 'layer_geom_type_s', literal('Image')
+
+to_field 'layer_geom_type_s', mods_xpath('mods:extension[@displayLabel="geo"]//dc:type'), first_only do |record, accumulator|
+  accumulator.flatten!.map!(&:text).map! { |v| v.gsub('Dataset#', '') }
+end
+to_field 'layer_geom_type_s' do |record, accumulator, context|
+  next if context.output_hash['layer_geom_type_s']
```

Comment:
Note: we will expand this further when someone implements https://github.com/sul-dlss/earthworks/issues/593 to support `content_type="file"`